### PR TITLE
[expo-barcod-scanner] Return bounding box

### LINKED ISSUE #2700: `onBarCodeScanned` result doesn't contain bounds of the barcodes
screenshot 2018-11-13 at 8 50 30 pm

img_5036

### LINKED ISSUE #8754: [iOS] BarcodeScanner should include bounds of detected barcodes
### Summary of Issue (just a few sentences)
When using the BarCodeScanner component to scan for QRcode, onBarCodeScanned callback on Android returns bounding data (coordinates of detected QRcode). But on iOS, it doesn't return bounding data.
Because in the documentation, it's not mentioned that bounds are available on Android only, I think it's a bug or a missing feature that I think we really need. The data of bounds is very important to make the scanner become more convenient and it can also improve user experience.

### Environment - output of `expo diagnostics` & the platform(s) you're targeting
iOS

### Steps to Reproduce
- Add BarCodeScanner component into React Native.
- Set onBarCodeScanned for BarCodeScanner.
- onBarCodeScanned  = (data)  => {
   console.log(data); // on IOS, it does not include bounds inside data
}
It's mentioned here: https://github.com/expo/expo/issues/2700

### Expected Behavior vs Actual Behavior
onBarCodeScanned  should return data including bounds.

Thanks in advance.

### PR DESCRIPTION
**Pull request by lukmccall**
# Why

Fixes https://github.com/expo/expo/issues/8754 and https://github.com/expo/expo/issues/2700.

# How

- Readd the bounding box information on Android (previously added by https://github.com/expo/expo/pull/2904).
- Add the same functionality to iOS.
- Update NCL and test-suite

Unfortunately, currently, we don't always return this information on iOS. For some barcode types, there is a problem with converting point coordinates from one space to another. 

# Test Plan

Tested using updated NCL and test-suite:
- iOS ✅
- Android ✅

### CODE REVIEW

**Comment by arealmaas** [comment]
😻 This is really needed. I know this is a difficult question to answer, but when would you expect this to be out live in expo? 🙏

**Comment by lukmccall** [comment]
> I know this is a difficult question to answer, but when would you expect this to be out live in expo? 🙏

Yes, it's a pretty difficult question :/ Unfortunately, it will be available in the `SDK 39` (probably).

File: docs/pages/versions/unversioned/sdk/bar-code-scanner.md [hunk 1]
```diff
@@ -130,3 +130,39 @@ Scan bar codes from the image given by the URL.
...
+## Types
+
+### `BarCodeScanner.BarCodePoint`
+
+Object of type `BarCodePoint` contains following keys:
+
+- **x (_number_)** -- The x value.
+- **y (_number_)** -- The y value.
+
+Those coordinates are represented in the coordinate space of the barcode source. For example, when you are using the bar code scanner view, those values will be adjusted to this space.
```
Comment (bbarthec) [inline docs/pages/versions/unversioned/sdk/bar-code-scanner.md:143]:
```suggestion
Those coordinates are represented in the coordinate space of the barcode source (e.g. when you are using the barcode scanner view, these values are adjusted to the dimensions of the view).
```

File: packages/expo-barcode-scanner/ios/EXBarCodeScanner/EXBarCodeScannerModule.m [hunk 2]
```diff
@@ -80,10 +80,7 @@ - (NSDictionary *)constantsToExport
       
       NSMutableArray *result = [NSMutableArray arrayWithCapacity:1];
       for (CIQRCodeFeature *feature in features)  {
-        [result addObject:@{
-                            @"type" : AVMetadataObjectTypeQRCode,
-                            @"data" : feature.messageString
-                            }];
+        [result addObject:[EXBarCodeScannerUtils ciqeCodeFeatureToDicitionary:feature barCodeType:AVMetadataObjectTypeQRCode]];
```
Comment (bbarthec) [inline packages/expo-barcode-scanner/ios/EXBarCodeScanner/EXBarCodeScannerModule.m:83]:
```suggestion
        [result addObject:[EXBarCodeScannerUtils ciQRCodeFeatureToDicitionary:feature barCodeType:AVMetadataObjectTypeQRCode]];
```

File: packages/expo-barcode-scanner/ios/EXBarCodeScanner/Utilities/EXBarCodeScannerUtils.h [hunk 3]
```diff
@@ -9,4 +10,8 @@
 + (AVCaptureDevice *)deviceWithMediaType:(AVMediaType)mediaType
                       preferringPosition:(AVCaptureDevicePosition)position;
 
++ (NSDictionary *)ciqeCodeFeatureToDicitionary:(CIQRCodeFeature *)barCodeScannerResult barCodeType:(NSString *)type;
```
Comment (bbarthec) [inline packages/expo-barcode-scanner/ios/EXBarCodeScanner/Utilities/EXBarCodeScannerUtils.h:13]:
```suggestion
+ (NSDictionary *)ciQRCodeFeatureToDicitionary:(CIQRCodeFeature *)barCodeScannerResult barCodeType:(NSString *)type;
```

File: packages/expo-barcode-scanner/ios/EXBarCodeScanner/Utilities/EXBarCodeScannerUtils.m [hunk 4]
```diff
@@ -46,4 +46,99 @@ + (AVCaptureDevice *)deviceWithMediaType:(AVMediaType)mediaType
   return [AVCaptureDevice defaultDeviceWithDeviceType:AVCaptureDeviceTypeBuiltInWideAngleCamera mediaType:mediaType position:position];
 }
 
++ (NSDictionary *)ciqeCodeFeatureToDicitionary:(CIQRCodeFeature *)barCodeScannerResult barCodeType:(NSString *)type
```
Comment (bbarthec) [inline packages/expo-barcode-scanner/ios/EXBarCodeScanner/Utilities/EXBarCodeScannerUtils.m:49]:
```suggestion
+ (NSDictionary *)ciQRCodeFeatureToDicitionary:(CIQRCodeFeature *)barCodeScannerResult barCodeType:(NSString *)type
```

**Review by bbarthec — APPROVED** [review]
LGTM 👍

**Comment by EvanBacon** [comment]
@arealmaas we release quarterly so it should be released end of September.

**Comment by enochj** [comment]
Will this be implemented for expo-camera as well?

**Comment by bbarthec** [comment]
@enochj, please create a `feature request issue` for this feature 😉

**Comment by adityar15** [comment]
is this implemented in sdk 42? For ios I am getting bounding box points in decimals and less than 1

**Comment by adityar15** [comment]
@bbarthec Thanks for the reply will do

**Pull request merged**